pkg/utils/utils.go

```diff
@@ -449,3 +457,164 @@ func DoesMigrationScheduleMigrateNamespaces(migrationSchedule stork_api.Migratio
 	}
 	return found, nil
 }
+
+// get Pod from PVC details
+func GetPodFromPVC(pvcName, namespace string) (*v1.Pod, error) {
+	fn := "GetPodFromPVCName"
+	// create pod list to store the relevant pods
+	var podList []*v1.Pod
+	pods, err := core.Instance().GetPodsUsingPVC(pvcName, namespace)
+	if err != nil {
+		errMsg := fmt.Sprintf("error fetching pods using PVC %s/%s: %v", namespace, pvcName, err)
+		logrus.Errorf("%s: %v", fn, errMsg)
+		return nil, fmt.Errorf(errMsg)
+	}
+	// filter out the pods that are created by us
+	for _, pod := range pods {
+		labels := pod.ObjectMeta.Labels
+		if _, ok := labels[kdmpDriver.DriverNameLabel]; ok {
+			continue
+		}
+		podList = append(podList, &pod)
+	}
+
+	if len(podList) == 0 {
+		errMsg := fmt.Sprintf("no application pod found for PVC %s/%s", namespace, pvcName)
+		logrus.Debugf("%s: %v", fn, errMsg)
+		return nil, fmt.Errorf(errMsg)
+	} else if len(podList) > 1 {
+		logrus.Warnf("%s: multiple application pods found using common PVC %s/%s", fn, namespace, pvcName)
+		// TODO we need to handle this case of PVC being used by multiple pods in the same namespace
+		// For now we will return the first pod in the list as a placeholder
+	}
+	return podList[0], nil
+}
+
+func getIdFromSecurityContext(securityContext interface{}) (int64, int64) {
+	uid := UndefinedId
+	gid := UndefinedId
+
+	// Baseline or privilege PSA can have nil security Context.
+	if securityContext == nil {
+		return uid, gid
+	}
+	switch s := securityContext.(type) {
+	case *v1.SecurityContext:
+		if s != nil {
```

Comment:
@prashanthpx I was also earlier of the same opinion , but I found it didn't landed in default case rather panicked, since I was accessing s.RunUser, So I added that check to fix issue during unit testing